### PR TITLE
Sticky Comments

### PR DESCRIPTION
**Pull request by danielgsims**
See #116

### CODE REVIEW

**Comment by danielgsims** [comment]
sticky-comments

File: src/Entity/Submission.php [hunk 1]
```diff
@@ -231,17 +231,36 @@ public function getComments(): Collection {
      */
     public function getTopLevelComments(): array {
         $criteria = Criteria::create();
-        $criteria->where(Criteria::expr()->isNull('parent'));
+	$criteria->where(Criteria::expr()->isNull('parent'))
```
Comment (nationwide13) [inline src/Entity/Submission.php]: fix indentation please

File: src/Entity/Submission.php [hunk 2]
```diff
@@ -231,17 +231,36 @@ public function getComments(): Collection {
...
+		 ->andWhere(Criteria::expr()->eq('stickied', false));
 
         $comments = $this->comments->matching($criteria)->toArray();
 
+
```
Comment (nationwide13) [inline src/Entity/Submission.php]: remove extra whitespace please

File: src/Entity/Submission.php [hunk 3]
```diff
@@ -231,17 +231,36 @@ public function getComments(): Collection {
...
     }
 
+    /**
+     * Get stickied comments,
+     *
+     * @return Comment[]
+     */
+    public function getStickyComments(): array {
+        $criteria = Criteria::create();
+	$criteria->where(Criteria::expr()->isNull('parent'))
```
Comment (nationwide13) [inline src/Entity/Submission.php]: fix indentation please

File: src/Entity/Submission.php [hunk 4]
```diff
@@ -231,17 +231,36 @@ public function getComments(): Collection {
...
+		 ->andWhere(Criteria::expr()->eq('stickied', true));
+
+        $comments = $this->comments->matching($criteria)->toArray();
+
+        return $comments;
+    }
+
+
+
```
Comment (nationwide13) [inline src/Entity/Submission.php]: remove extra whitespace please

File: src/Migrations/Version20180331122927.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * Auto-generated Migration: Please modify to your needs!
+ */
+class Version20180331122927 extends AbstractMigration
+{
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'postgresql', 'Migration can only be executed safely on \'postgresql\'.');
+
+	$this->addSql('ALTER TABLE comments ADD COLUMN "stickied" BOOLEAN NOT NULL DEFAULT FALSE');
```
Comment (nationwide13) [inline src/Migrations/Version20180331122927.php]: fix indentation please

File: src/Migrations/Version20180331122927.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    public function down(Schema $schema)
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'postgresql', 'Migration can only be executed safely on \'postgresql\'.');
+	$this->addSql('ALTER TABLE comments DROP COLUMN "stickied"');
```
Comment (nationwide13) [inline src/Migrations/Version20180331122927.php]: fix indentation please

File: assets/less/comments.less [hunk 7]
```diff
@@ -53,3 +53,12 @@
...
 }
+
+.comment-info .comment-sticky {
+  background: @comment-sticky-background-color;
+  border-radius: 2px;
+  color: @comment-sticky-color;
+  padding: 0 0.25rem;
+}
+
+
```
Comment (nationwide13) [inline assets/less/comments.less]: remove extra whitespace please

**Comment by danielgsims** [comment]
@nationwide13 - Thanks for checking the formatting, I'll make sure to clear it up. I'm on a new machine and don't have a proper editor set up yet.

**Comment by danielgsims** [comment]
@nationwide13 - I think it's good to go for moderators. I have one outstanding todo which is to let moderators or admins perform the action, without the admin role IsGranted check superseding the forum mod check.

If anyone knows an example in the code I could look at, I can copy that over.

**Pull request merged**